### PR TITLE
Add jabber layer

### PR DESCRIPTION
**Pull request by toshism**
Not sure how popular jabber is, but I use it.

I'm open to suggestions for keybindings.

### CODE REVIEW

File: contrib/jabber/packages.el [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+(defun jabber/init-jabber ()
+  "Initialize jabber"
+  (use-package jabber
+    :defer t
+    :init
+    (evil-leader/set-key
+      "aj" 'jabber-connect-all)
+    :config
+    (progn
+      (evil-leader/set-key-for-mode 'jabber-roster-mode
```
Comment (trishume) [inline contrib/jabber/packages.el]: You should put these keybindings under the `SPC m` prefix that is designated for mode-specific bindings.

**Comment by kini** [comment]
Are you OK to use that logo, BTW? In any case, Jabber was renamed XMPP many years ago (despite this package still being called "jabber" on MELPA). [The XMPP logo](https://xmpp.org/images/xmpp.svg) has clear licensing terms, which you can read [here](https://xmpp.org/about-xmpp/faq/#logo). I couldn't find any info about the Jabber logo or who owns it. I know there was some sort of legal issue involved in the renaming of Jabber to XMPP back in the day, so maybe it's a good idea to call everything "XMPP" when possible...?

**Comment by toshism** [comment]
It looks like the jabber logo is [CC licensed](http://commons.wikimedia.org/wiki/File:Jabber_logo.png) now. It was previously listed as gpl on wikimedia. Perhaps it needs attribution? I am far from an expert on  the various CC licenses. 

It would seem a bit odd to me to name the layer XMPP when the mode is jabber? I don't have a strong opinion on it, but I would lean towards jabber personally.

**Comment by trishume** [comment]
Yah I would name it Jabber and then make sure to mention XMPP somewhere for searchability. Better to name things by package when in doubt because it makes it easy to find the layer for a given package.

**Comment by kini** [comment]
Added an attribution at toshism/spacemacs#1

**Comment by syl20bnr** [comment]
Thank you !
Cherry-picked into `develop` branch, you can safely delete your branch.

**Pull request closed without merging**